### ISSUE TITLE
Converted Onnx model from gpt2 does not have the same input shape

### ISSUE THREAD

**Issue by farazk86** [microsoft/onnxruntime#6271]
Hi,

I want to convert the ``distilgpt2`` model from transformers to onnx model. and then want to convert this onnx to tflite. The problem is that the input shapes are different at all levels, so I must be doing something wrong.

I'm using 

```python
python -m onnxruntime_tools.transformers.convert_to_onnx -m [path_to_gpt2_pytorch_model_directory] --output quantized.onnx 
```

as provided here: https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/transformers

I'm using the following code

```python
!git clone --depth 1 --branch v2.11.0 https://github.com/huggingface/transformers.git
!pip install -q ./transformers

!pip install onnxruntime
!pip install onnxruntime-tools onnx-tf==1.6.0 tensorflow-addons==0.11.2

import onnxruntime_tools 

!python -m onnxruntime_tools.transformers.convert_to_onnx \
-m /content/drive/MyDrive/gpt2_finetuned_models/checkpoint-20000/ \
--output /content/onnx_model/model.onnx

import onnxruntime

sess = onnxruntime.InferenceSession ('/content/onnx_model/model.onnx')
# Let’s see the input name and shape.

input_name = sess.get_inputs()[0].name
print("input name", input_name)
input_shape = sess.get_inputs()[0].shape
print("input shape", input_shape)
input_type = sess.get_inputs()[0].type
print("input type", input_type)
```

The above converts the output without any problems but the inference of onnx model generates the following outputs:

```
input name input_ids
input shape ['batch_size', 'seq_len']
input type tensor(int64)
```

Here, what is the value of ``batch_size`` and ``seq_len`` ?

Because when I convert this to tflite and inference of the converted tflite model produces the following output:

```
== Input details ==
name: input_ids
shape: [1 1]
type: <class 'numpy.int64'>

== Output details ==
name: logits
shape: [    1     1 50257]
type: <class 'numpy.float32'>
```

shape ``[1 1]`` is wrong, it should be ``[1 128]`` 

Why are these different? 

What am I doing wrong?

Thanks

**Comment by tianleiwu** [comment]
batch_size and seq_len (sequence length) are dynamic axis for input shape. For example, input_ids with shape [1, 128] has batch_size=1 and seq_len=128. Since your input has seq_len=1, I think the output shape [1, 1, 50257] is correct. Could you create an input_ids with shape [1, 128] and try again?

**Comment by farazk86** [comment]
> batch_size and seq_len (sequence length) are dynamic axes for input shape. For example, input_ids with shape [1, 128] has batch_size=1 and seq_len=128. Since your input has seq_len=1, I think the output shape [1, 1, 50257] is correct. Could you create an input_ids with shape [1, 128] and try again?

Thank you for your reply.

Can you please let me know how I can set the values for ``input_ids`` in the model? As mentioned above, I am using

```python
!python -m onnxruntime_tools.transformers.convert_to_onnx \
-m /content/drive/MyDrive/gpt2_finetuned_models/distilltwosentencehorror/checkpoint-20000/ \
--output /content/onnx_model/model.onnx
```

to convert the gpt2 model to onnx, would ``input_ids`` be set after conversion?

Thanks

**Comment by farazk86** [comment]
Hi @tianleiwu 

apologies if this is a very basic question but I still cant set the ``input_ids`` parameter, Can you please let me know how I can set that and hopefully then be able to export to a working ``tflite`` model.

Thank you

**Comment by farazk86** [comment]
Hi @tianleiwu 

So an update on this.. I managed to set the input id shapes as advised by you. I did this at the tflite model conversion stage:

```python
converter = tf.compat.v1.lite.TFLiteConverter.from_frozen_graph('tf_model.pb',
                                                                input_arrays=['input_ids'],
                                                                output_arrays=['logits'],
                                                                input_shapes={'input_ids': [1, 64]}
                                                                )
```

But the tflite inference does not follow the input shape.. following the same script as in my above post, the output generated is:

```
== Input details ==
name: input_ids
shape: [ 1 64]
type: <class 'numpy.int64'>
== Output details ==
name: logits
shape: [    1     1 50257]
type: <class 'numpy.float32'>
```

my inference script is simply invoking the tflite model:

```python
interpreter = tf.lite.Interpreter(model_path="model.tflite")
interpreter.allocate_tensors()

# Get input and output tensors.
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

print("== Input details ==")
print("name:", input_details[0]['name'])
print("shape:", input_details[0]['shape'])
print("type:", input_details[0]['dtype'])

print("\n== Output details ==")
print("name:", output_details[0]['name'])
print("shape:", output_details[0]['shape'])
print("type:", output_details[0]['dtype'])
```
